File: zero-click-budgeting/api/src/services/budgetEngine.js
export function computeGauge(transactions = [], limits = { monthly: 1 }) {
  const total = transactions.reduce((sum, t) => sum + (Number(t.amount) || 0), 0);
  const monthly = limits.monthly || 1;
  const percent = Math.min(100, Math.round((100 * total) / monthly));
  const status = percent < 70 ? 'green' : percent < 90 ? 'orange' : 'red';
  return { percent, status, monthly_limit: monthly, spend: total };
}

File: zero-click-budgeting/budget-engine/src/budget_engine.py
def compute_gauge(transactions, limits):
    total_spend = sum(float(t.get('amount', 0) or 0) for t in transactions)
    monthly_limit = limits.get('monthly', 1) or 1
    pct = min(100, round(100 * total_spend / monthly_limit))
    status = 'green' if pct < 70 else 'orange' if pct < 90 else 'red'
    return { 'percent': pct, 'status': status, 'monthly_limit': monthly_limit, 'spend': total_spend }